martin/src/cog/source.rs

```diff
@@ -521,4 +560,34 @@ mod tests {
         "###);
         assert_yaml_snapshot!(transformation, @"~");
     }
+
+    #[rstest]
+    #[case(
+        Some(vec![0.0, 0.0, 0.0, 1_620_750.250_8, 4_277_012.715_3, 0.0]),None,
+        [1_620_750.250_8, 4_277_012.715_3, 0.0]
+    )]
+    #[case(
+        None,Some(vec![
+            0.0, 100.0, 0.0, 400_000.0, 100.0, 0.0, 0.0, 500_000.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,
+            0.0, 1.0,
+        ]),
+        [400_000.0, 500_000.0, 0.0]
+    )]
+    fn can_get_origin(
+        #[case] tie_point: Option<Vec<f64>>,
+        #[case] matrix: Option<Vec<f64>>,
+        #[case] expected: [f64; 3],
```

Comment:
Not nesssesary, but I think adding a test cases for the error path would be very nice ^^

```suggestion
        #[case] expected: Option<[f64; 3]>,
```